### PR TITLE
[Android][vcpkg] Compile android with vcpkg

### PR DESCRIPTION
**Pull request by talregev**
Compile android with vcpkg.

### CODE REVIEW

File: .github/workflows/android.yml [hunk 1]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        type: [manager, libs, apps]
+        type: [manager, libs, apps, vcpkg_manager, vcpkg_libs, vcpkg_apps]
```
Comment (AenBleidd) [inline .github/workflows/android.yml:16]: `vcpkg_manager` -> `manager-vcpkg`
`vcpkg_libs` -> `libs-vcpkg`
`vcpkg_apps` -> `apps-vcpkg`

File: .github/workflows/android.yml [hunk 2]
```diff
@@ -47,8 +47,20 @@ jobs:
...
+        if: ${{ success() && matrix.type == 'vcpkg_libs' }}
+        run: ./android/ci_build_vcpkg_libraries.sh
+
+      - name: Build vcpkg apps
+        if: ${{ success() && matrix.type == 'vcpkg_apps' }}
+        run: ./android/ci_build_vcpkg_apps.sh
+
       - name: Deploy to BinTray
-        if: ${{ success() && ! contains(matrix.type, 'libs') }}
+        if: ${{ success() && ! contains(matrix.type, 'libs') && ! contains(matrix.type, 'vcpkg_libs') }}
```
Comment (AenBleidd) [inline .github/workflows/android.yml:63]: looks redundant:
`! contains(matrix.type, 'libs')` should already filter out both plans that contains `libs` it their names

File: .github/workflows/android.yml [hunk 3]
```diff
@@ -57,7 +69,7 @@ jobs:
 
       - name: Upload artifacts
         uses: actions/upload-artifact@v2
-        if: ${{ ! contains(matrix.type, 'libs') }}
+        if: ${{ ! contains(matrix.type, 'libs') && ! contains(matrix.type, 'vcpkg_libs') }}
```
Comment (AenBleidd) [inline .github/workflows/android.yml:72]: The same here

File: android/build_boinc_arm.sh [hunk 4]
```diff
@@ -23,14 +23,26 @@ export TOOLCHAINROOT="$NDK_ROOT/toolchains/llvm/prebuilt/linux-x86_64/"
...
+export VCPKG_DIR=$VCPKG_ROOT/installed/arm-android
+
+VCPKG_CONFIG_FLAG=""
+VCPKG_LDFLAG=""
+
+if [ $BUILD_WITH_VCPKG = "yes" ]; then
+    VCPKG_LDFLAG="-L$VCPKG_DIR/lib"
+    VCPKG_CONFIG_FLAG="--with-libcurl=$VCPKG_DIR --with-ssl=$VCPKG_DIR --enable-vcpkg"
+else
+    VCPKG_CONFIG_FLAG="--with-ssl=$TCINCLUDES"
```
Comment (AenBleidd) [inline android/build_boinc_arm.sh:35]: You use variable with `VCPKG_` prefix for builds without vcpkg, it's misleading, and should be changed.
It's the same for other similar scripts, please change them also, I don't want to publish the same comment in any file

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: .github/workflows/android.yml (hunk 1 above)
Comment (talregev) [reply .github/workflows/android.yml]: fix

File: .github/workflows/android.yml (hunk 2 above)
Comment (talregev) [reply .github/workflows/android.yml]: fix

File: .github/workflows/android.yml (hunk 3 above)
Comment (talregev) [reply .github/workflows/android.yml]: fix

File: android/build_boinc_arm.sh (hunk 4 above)
Comment (talregev) [reply android/build_boinc_arm.sh]: fix

File: android/build_boinc_arm.sh [hunk 5]
```diff
@@ -23,14 +23,26 @@ export TOOLCHAINROOT="$NDK_ROOT/toolchains/llvm/prebuilt/linux-x86_64/"
...
+
+CONFIG_FLAGS=""
+VCPKG_LDFLAG=""
+
+if [ $BUILD_WITH_VCPKG = "yes" ]; then
+    VCPKG_LDFLAG="-L$VCPKG_DIR/lib"
+    CONFIG_FLAGS="--with-libcurl=$VCPKG_DIR --with-ssl=$VCPKG_DIR --enable-vcpkg"
+else
+    CONFIG_FLAGS="--with-ssl=$TCINCLUDES"
+    VCPKG_LDFLAG="-L$TCSYSROOT/usr/lib -L$TCINCLUDES/lib"
```
Comment (AenBleidd) [inline android/build_boinc_arm.sh:36]: The same applicable to this variable: `VCPKG_LDFLAG`
Maybe it's better to rename to `CONFIG_LDFLAGS`?

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: android/build_boinc_arm.sh (hunk 5 above)
Comment (talregev) [reply android/build_boinc_arm.sh]: fix

**Review by AenBleidd — APPROVED** [review]
LGTM

File: .github/workflows/android.yml [hunk 6]
```diff
@@ -13,7 +13,7 @@ jobs:
     runs-on: ubuntu-latest
     strategy:
       matrix:
-        type: [manager, libs, apps]
+        type: [manager, libs, apps, vcpkg-manager, vcpkg-libs, vcpkg-apps]
```
Comment (AenBleidd) [inline .github/workflows/android.yml:16]: `vcpkg-manager` -> `manager-vcpkg`
`vcpkg-libs` -> `libs-vcpkg`
`vcpkg-apps` -> `apps-vcpkg`

**Review by AenBleidd — CHANGES_REQUESTED** [review]

File: .github/workflows/android.yml (hunk 6 above)
Comment (talregev) [reply .github/workflows/android.yml]: fix

**Review by AenBleidd — APPROVED** [review]
LGTM

**Pull request merged**